### ISSUE TITLE
Failed when generate RAML from my Jaxrs codes

### ISSUE THREAD

**Issue by hantsy** [mulesoft-labs/raml-for-jax-rs#192]
The exception:

```
spoon.compiler.ModelBuildingException: The method getId() is undefined for the type PostDetail at E:\hantsylabs\angularjs-ee7-sample\sample\blog-api\src\main\java\com\hantsylab\example\ee7\blog\api\PostResource.java:62
    at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.report(JDTBasedSpoonCompiler.java:535)
    at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.reportProblems(JDTBasedSpoonCompiler.java:551)
    at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:507)
    at spoon.Launcher.buildModel(Launcher.java:750)
    at spoon.Launcher.run(Launcher.java:701)
    at org.raml.jaxrs.codegen.maven.JaxrsRamlCodegenMojo.execute(JaxrsRamlCodegenMojo.java:139)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher
```

I have used Lombok project to generate some codes in the POJOs, but seem the maven plugin can not get the method from generated classes.

My maven config.

```
<plugin>

                    <groupId>org.raml.plugins</groupId>
                    <artifactId>jaxrs-raml-maven-plugin</artifactId>
                    <version>1.3.4</version>
                    <configuration>
                        <sourceDirectory>${basedir}/src/main/java</sourceDirectory>
                        <outputFile>${basedir}/target/generated-sources/jaxrs-raml/api.raml</outputFile>
                        <basePackageName>com.hantsylab.example.ee7.blog.api</basePackageName>
                        <jaxrsVersion>2.0</jaxrsVersion>
                        <useJsr303Annotations>true</useJsr303Annotations>
                        <jsonMapper>jackson2</jsonMapper>
                        <removeOldOutput>true</removeOldOutput>
                    </configuration>
                    <executions>
                        <execution>
                            <goals>
                                <goal>generate-raml</goal>
                            </goals>
                            <phase>test</phase>
                        </execution>
                    </executions>
                </plugin>
```

**Comment by gkozyryatskyy** [comment]
+1

**Comment by clovis-guillemot** [comment]
+1 
any solution for 8 months @hantsy ?

**Comment by jstoiko** [comment]
@clovis-guillemot: there is a new version in the works: https://github.com/mulesoft/raml-for-jax-rs/tree/release/2.0.0 and it's currently released as `2.0.0-RC1` on `repository-master.mulesoft.org`.

It's a complete refactoring with support for RAML 1.0. The docs along with examples for generating RAML from JAXRS, i.e.`jaxrs-to-raml`, is [here](https://github.com/mulesoft/raml-for-jax-rs/blob/release/2.0.0/jaxrs-to-raml/README.md). 

Would love to hear your feedback. Let me know if you have any questions.
